example(template-syntax): follow style-guide and other updates

chalin opened the pull request:
- Follow style guide: rename `sizer.component.ts` to `my-sizer.component.ts` (given the selector `my-sizer`); also appropriately rename class.
- Rename `AppComponent.fontSize` to `AppComponent.fontSizePx` because it is assigned to `[style.font-size.px]` and so represents a size in px, not a general font-size value (which could be, e.g., “x-large”).
- Added small interactive example of `NgStyle`.

wardbell commented on public/docs/_examples/template-syntax/e2e-spec.ts:
```diff
@@ -31,14 +31,13 @@ describe('Template Syntax', function () {
+    let incButton = div.element(by.buttonText('+'));
+    let input = div.element(by.css('input'));
+    let initSize = await input.getAttribute('value');
+    incButton.click();
```
Are you sure that we don't have to wait for the browser to settle after clicking the button. That line had a purpose at one time.

chalin replied:
Yes, e2e tests run fine:
```
Suites passed:
  public/docs/_examples/template-syntax/ts
All tests passed
```

wardbell commented on public/docs/_examples/template-syntax/ts/app/app.component.html:
```diff
@@ -351,21 +351,22 @@
   <!-- #docregion two-way-1 -->
-  <my-sizer [(size)]="fontSize"></my-sizer>
-  <div [style.font-size.px]="fontSize">Resizable Text</div>
+  <my-sizer [(size)]="fontSizePx"></my-sizer>
```
This change seems picayune.  I feel it distracts from the point and starts a conversation that isn't helpful.  I can understand that you feel that being inspecific about the units in the property name provokes a distracting conversation too. I don't actually feel strongly about this so I'd say "whatever, OK" .

chalin replied:
Actually, this is mainly to avoid a conflict with a string property named `fontSize` whose initial value is `"large"`.

It seemed to me that since the property is being assigned to `[style.font-size.px]` (see the old line 355 above) it could naturally be named `fontSizePx` (which is the camel-case equivalent of the style attribute it is being assigned to).

wardbell replied:
I'm cool with it

wardbell commented on public/docs/ts/latest/guide/template-syntax.jade:
```diff
@@ -490,8 +490,8 @@ table
     See the API reference for
-    [viewChild](../api/core/index/ViewChild-decorator.html) and
-    [contentChild](../api/core/index/ContentChild-decorator.html).
+    [ViewChild](../api/core/index/ViewChild-decorator.html) and
```
Good

chalin replied:
👍

wardbell commented on public/docs/ts/latest/guide/template-syntax.jade:
```diff
@@ -778,7 +775,7 @@ block style-property-name-dart-diff
   <a id="eventemitter"></a>
   <a id="custom-event"></a>
-  ### Custom Events with EventEmitter
+  ### Custom events with EventEmitter
```
Please make "EventEmitter" italic. That's the (desired) standard presentation of code elements in a header ... because `EventEmitter` doesn't display well in a header.

wardbell commented on public/docs/ts/latest/guide/template-syntax.jade:
```diff
@@ -853,36 +850,44 @@ block style-property-name-dart-diff
   The `[(x)]` syntax is easy to demonstrate when the element has a settable property called `x`
   and a corresponding event named `xChange`. 
-  Here's a `SizerComponent` that fits the pattern.
+  Here's a `MySizerComponent` that fits the pattern.
```
See my corrective above

chalin replied:
Reverted.

chalin commented:
@wardbell - addressed your concerns with the last commit:
- Keep `my-` and `my` prefixes on selectors (for components and
directives respectively).
- Drop `my-` from file names.
- Drop `My` as component class prefix.
